migrator/src/main/scala/com/scylladb/migrator/alternator/DynamoDBInputFormat.scala

```diff
@@ -0,0 +1,86 @@
+package com.scylladb.migrator.alternator
+import org.apache.commons.logging.LogFactory
+import org.apache.hadoop.dynamodb.DynamoDBConstants
+import org.apache.hadoop.mapred.{ InputSplit, JobConf }
+
+/**
+  * Specializes the split strategy:
+  *    - do not bound the maximum number of partitions by the available memory per node
+  *    - use as many partitions as the number of scan segments
+  *    - by default, create segments that split the data into 128 MB chunks
+  */
+class DynamoDBInputFormat extends org.apache.hadoop.dynamodb.read.DynamoDBInputFormat {
```

Comment:
Our class overrides the base implementation of the methods `getSplits` and `getNumSegments`. Our implementation is based on the original implementation available here:

https://github.com/awslabs/emr-dynamodb-connector/blob/07391dd0937bdbb20b86ec79444798df11b0711f/emr-dynamodb-hadoop/src/main/java/org/apache/hadoop/dynamodb/read/AbstractDynamoDBInputFormat.java

But with some variations.